rust/src/dataset.rs

```diff
@@ -954,6 +955,25 @@ impl Dataset {
         read_manifest_indexes(&self.object_store, &manifest_file, &self.manifest).await
     }
 
+    pub async fn statistics(&self, index_name: String) -> Result<String> {
```

Comment:
Maybe this should return `Ok(None)` if the index with that name isn't found?

Also, just like in Python, this should be more specific since we will add other kinds of statistics to datasets.
```suggestion
    pub async fn index_statistics(&self, index_name: String) -> Result<Option<String>> {
```